component/azstorage/config.go

```diff
@@ -124,24 +124,25 @@ const DefaultMaxResultsForList int32 = 2
 // https://github.com/Azure/go-autorest/blob/a46566dfcbdc41e736295f94e9f690ceaf50094a/autorest/adal/token.go#L788
 // newServicePrincipalTokenFromMSI : reads them directly from env
 const (
-	EnvAzStorageAccount                = "AZURE_STORAGE_ACCOUNT"
-	EnvAzStorageAccountType            = "AZURE_STORAGE_ACCOUNT_TYPE"
-	EnvAzStorageAccessKey              = "AZURE_STORAGE_ACCESS_KEY"
-	EnvAzStorageSasToken               = "AZURE_STORAGE_SAS_TOKEN"
-	EnvAzStorageIdentityClientId       = "AZURE_STORAGE_IDENTITY_CLIENT_ID"
-	EnvAzStorageIdentityResourceId     = "AZURE_STORAGE_IDENTITY_RESOURCE_ID"
-	EnvAzStorageIdentityObjectId       = "AZURE_STORAGE_IDENTITY_OBJECT_ID"
-	EnvAzStorageSpnTenantId            = "AZURE_STORAGE_SPN_TENANT_ID"
-	EnvAzStorageSpnClientId            = "AZURE_STORAGE_SPN_CLIENT_ID"
-	EnvAzStorageSpnClientSecret        = "AZURE_STORAGE_SPN_CLIENT_SECRET"
-	EnvAzStorageSpnOAuthTokenFilePath  = "AZURE_OAUTH_TOKEN_FILE"
-	EnvAzStorageAadEndpoint            = "AZURE_STORAGE_AAD_ENDPOINT"
-	EnvAzStorageAuthType               = "AZURE_STORAGE_AUTH_TYPE"
-	EnvAzStorageBlobEndpoint           = "AZURE_STORAGE_BLOB_ENDPOINT"
-	EnvAzStorageAccountContainer       = "AZURE_STORAGE_ACCOUNT_CONTAINER"
-	EnvAzAuthResource                  = "AZURE_STORAGE_AUTH_RESOURCE"
-	EnvAzStorageCpkEncryptionKey       = "AZURE_STORAGE_CPK_ENCRYPTION_KEY"
-	EnvAzStorageCpkEncryptionKeySha256 = "AZURE_STORAGE_CPK_ENCRYPTION_KEY_SHA256"
+	EnvAzStorageAccount                  = "AZURE_STORAGE_ACCOUNT"
+	EnvAzStorageAccountType              = "AZURE_STORAGE_ACCOUNT_TYPE"
+	EnvAzStorageAccessKey                = "AZURE_STORAGE_ACCESS_KEY"
+	EnvAzStorageSasToken                 = "AZURE_STORAGE_SAS_TOKEN"
+	EnvAzStorageIdentityClientId         = "AZURE_STORAGE_IDENTITY_CLIENT_ID"
+	EnvAzStorageIdentityResourceId       = "AZURE_STORAGE_IDENTITY_RESOURCE_ID"
+	EnvAzStorageIdentityObjectId         = "AZURE_STORAGE_IDENTITY_OBJECT_ID"
+	EnvAzStorageSpnTenantId              = "AZURE_STORAGE_SPN_TENANT_ID"
+	EnvAzStorageSpnClientId              = "AZURE_STORAGE_SPN_CLIENT_ID"
+	EnvAzStorageSpnClientSecret          = "AZURE_STORAGE_SPN_CLIENT_SECRET"
+	EnvAzStorageSpnOAuthTokenFilePath    = "AZURE_OAUTH_TOKEN_FILE"
+	EnvAzStorageSpnWorkloadIdentityToken = "WORKLOAD_IDENTITY_TOKEN"
```

Comment:
if we use `WORKLOAD_IDENTITY_TOKEN` env var, who is responsible for token refresh?

Reply:
The token was generated by kubelet when it try to mount volume, and it will be passed to csi driver and csi driver set it as WORKLOAD_IDENTITY_TOKEN env. We should only use this token for mounting.

Reply:
`AZURE_OAUTH_TOKEN_FILE` is used here, it's for blobfuse mount, I think you also need to change the code here when WORKLOAD_IDENTITY_TOKEN env is provided explicitly by the user:

https://github.com/Azure/azure-storage-fuse/blob/42b3b19b42dbe36e5b37e7b4d81115c6a319b277/component/azstorage/azauthspn.go#L58-L66